inst/checklists/new_model_checklist.md

```diff
@@ -0,0 +1,31 @@
+* [ ] Create the main modeling function (e.g. `nearest_neighbors()`). 
+
+* [ ] Write the initial functions that will populate the parsnip model “database”. For example, `parsnip::set_mode()` and so on. [(example)](https://github.com/tidymodels/parsnip/blob/c54f07b7e1f7ce164aab8f95bc7b1356b68558c8/R/proportional_hazards_data.R)
+
+* [ ] _[not in parsnip]_ Put the `set_*()` calls in a wrapper function that will register the model with parsnip. [(example)](https://github.com/tidymodels/rules/blob/main/R/rule_fit_data.R#L1)
+
+* [ ] _[not in parsnip]_ Add the registration functions to `.onLoad()` in the `zzz.R` file. [(example)](https://github.com/tidymodels/rules/blob/main/R/zzz.R#L7:L18)
+
+* [ ] Create secondary methods such as `update()`, `tuneable()` and so on. 
```

Comment:
```suggestion
* [ ] Create an `update()` method.
```
I would separate the required `update()` method from the optional methods for `translate()`, `check_args()` and `tunable()`